src/snowflake/snowpark/_internal/packaging_utils.py

```diff
@@ -311,17 +315,20 @@ def pip_install_packages_to_target_folder(packages: List[str], target: str) -> N
         process = subprocess.Popen(
             pip_command + ["install", "-t", target, *packages],
             stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE,
             universal_newlines=True,
         )
-        process.wait()
+        stdout, stderr = process.communicate(timeout=timeout)
```

Comment:
The standard output (stdout) and standard error (stderr) are captured with subprocess.PIPE. However, the Popen call was earlier followed by process.wait(), which waits for the process to complete, but does not read the output. If the child process generates enough output to fill the OS pipe buffer, the child process will block waiting for the OS to make room in the pipe buffer. This will never occur because the parent process is waiting for the child process to exit before it reads the output.

This can be resolved by using process.communicate() instead of process.wait(). The communicate() function waits for the process to terminate and also reads all output, thereby avoiding the deadlock situation.

See details at https://docs.python.org/3/library/subprocess.html#subprocess.Popen.wait